django/contrib/sessions/backends/base.py

```diff
@@ -54,6 +57,12 @@ def __contains__(self, key):
         return key in self._session
 
     def __getitem__(self, key):
+        if key == LANGUAGE_SESSION_KEY:
+            warnings.warn(
+                "Storing the client language in the session is deprecated. "
```

Comment:
I would if a message  like this might be more intuitive?
"The client language will no longer be stored in `request.session` in Django 2.1. Read it from `request.COOKIES[django.utils.translation.LANGUAGE_COOKIE_NAME]` instead."